components/profile/index.test.jsx

```diff
@@ -33,10 +40,19 @@ describe("Profile", () => {
   });
 
   test("renders an editable profile", () => {
-    const { asFragment } = renderWithTheme(
-      <Profile profileIri={profileIri} editing />
-    );
-    expect(asFragment()).toMatchSnapshot();
+    const session = mockSession();
+    const SessionProvider = mockSessionContextProvider(session);
+    act(() => {
+      const { asFragment } = renderWithTheme(
+        <SessionProvider>
+          <Profile profileIri={profileIri} editing />
+        </SessionProvider>
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+    expect(screen.findByTestId(TESTCAFE_ID_NAME_FIELD)).not.toBeNull();
+    expect(screen.findByTestId(TESTCAFE_ID_ROLE_FIELD)).not.toBeNull();
+    expect(screen.findByTestId(TESTCAFE_ID_ORG_FIELD)).not.toBeNull();
```

Comment:
I don't quite have the context to fix this entirely, but if you try to make the code invalid, this test will still succeed. The `findBy*` functions return a Promise that either resolve successfully or reject, so if you want to test whether an element is present you should test for the Promise to resolve. This would require making the test function `async` and then doing [something like](https://jestjs.io/docs/asynchronous#resolves--rejects):

```suggestion
    await expect(screen.findByTestId(TESTCAFE_ID_NAME_FIELD)).resolves.not.toBeNull();
    await expect(screen.findByTestId(TESTCAFE_ID_ROLE_FIELD)).resolves.not.toBeNull();
    await expect(screen.findByTestId(TESTCAFE_ID_ORG_FIELD)).resolves.not.toBeNull();
```

Alternatively, it looks like the `queryBy*` functions do do what you expect:

```suggestion
    expect(screen.queryByTestId(TESTCAFE_ID_NAME_FIELD)).not.toBeNull();
    expect(screen.queryByTestId(TESTCAFE_ID_ROLE_FIELD)).not.toBeNull();
    expect(screen.queryByTestId(TESTCAFE_ID_ORG_FIELD)).not.toBeNull();
```

Unfortunately, when I apply the above, the tests start failing, but you're probably in a better position to fix that than I am.

Reply:
Great catch! This actually helped me find a bug I had missed 🐛